<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>A Computational Pipeline for Bulk ATAC-Seq Profiles • cinaR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="A Computational Pipeline for Bulk ATAC-Seq Profiles">
<meta name="description" content="Differential analyses and Enrichment pipeline for bulk ATAC-seq data analyses. This package combines different packages to have an ultimate package for both data analyses and visualization of ATAC-seq data. Methods are described in Karakaslar et al. (2021) &lt;doi:10.1101/2021.03.05.434143&gt;.">
<meta property="og:description" content="Differential analyses and Enrichment pipeline for bulk ATAC-seq data analyses. This package combines different packages to have an ultimate package for both data analyses and visualization of ATAC-seq data. Methods are described in Karakaslar et al. (2021) &lt;doi:10.1101/2021.03.05.434143&gt;.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">cinaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/cinaR.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/eonurk/cinaR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="cinar-">cinaR <a href="https://eonurk.github.io/cinaR/"><img src="reference/figures/cinaR.png" align="right" alt="" width="139"></a>
<a class="anchor" aria-label="anchor" href="#cinar-"></a>
</h1></div>
<!-- badges: start -->

<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p><code>cinaR</code> is a single wrapper function for end-to-end computational analyses of bulk ATAC-seq (or RNA-seq) profiles. Starting from a consensus peak file, it outputs differentially accessible peaks, enrichment results, and provides users with various configurable visualization options. For more details, please see the <a href="https://www.biorxiv.org/content/10.1101/2021.03.05.434143v2" class="external-link">preprint</a>.</p>
<p><img src="reference/figures/overview@5x.png"></p>
</div>
<div class="section level2">
<h2 id="llm-and-agent-usage">LLM and agent usage<a class="anchor" aria-label="anchor" href="#llm-and-agent-usage"></a>
</h2>
<p>For model-friendly discovery and tool integration:</p>
<ul>
<li>
<code>llms.txt</code> provides a compact machine-oriented index for <code>cinaR</code>.</li>
<li>
<code>inst/mcp/cinaR-mcp-tools.json</code> provides starter MCP tool definitions with JSON Schemas.</li>
</ul>
<p>Useful entry points for assistants:</p>
<ul>
<li>
<code><a href="reference/cinaR.html">cinaR()</a></code>: differential analysis + enrichment (+ optional TF activity) from consensus data.</li>
<li>
<code><a href="reference/prep_scATAC_cinaR.html">prep_scATAC_cinaR()</a></code>: pseudobulk 10x scATAC peak-by-cell matrices into cinaR-ready inputs.</li>
<li>
<code><a href="reference/prep_scATAC_seurat.html">prep_scATAC_seurat()</a></code>: same preprocessing directly from a Seurat/Signac object.</li>
</ul>
<p>Use the package site for detailed docs and examples: <a href="https://eonurk.github.io/cinaR/" class="uri">https://eonurk.github.io/cinaR/</a>.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># CRAN mirror</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"cinaR"</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="development-version">Development version<a class="anchor" aria-label="anchor" href="#development-version"></a>
</h3>
<p>To get bug fix and use a feature from the development version:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"eonurk/cinaR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="known-installation-issues">Known Installation Issues<a class="anchor" aria-label="anchor" href="#known-installation-issues"></a>
</h3>
<p>Sometimes bioconductor related packages may not be installed automatically.<br>
Therefore, you may need to install them manually:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ChIPseeker"</span>, <span class="st">"DESeq2"</span>, <span class="st">"edgeR"</span>, <span class="st">"fgsea"</span>,<span class="st">"GenomicRanges"</span>, <span class="st">"limma"</span>, <span class="st">"preprocessCore"</span>, <span class="st">"sva"</span>, <span class="st">"TxDb.Hsapiens.UCSC.hg38.knownGene"</span>, <span class="st">"TxDb.Hsapiens.UCSC.hg19.knownGene"</span>, <span class="st">"TxDb.Mmusculus.UCSC.mm10.knownGene"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/eonurk/cinaR/" class="external-link">cinaR</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; Checking for required Bioconductor packages...</span></span>
<span><span class="co">#&gt; All required Bioconductor packages are already installed.</span></span>
<span></span>
<span><span class="co"># create contrast vector which will be compared.</span></span>
<span><span class="va">contrasts</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, </span>
<span>              <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"B6"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span>, <span class="st">"NZO"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If reference genome is not set hg38 will be used!</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">bed</span>, <span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"mm10"</span><span class="op">)</span></span>
<span><span class="co">#&gt; &gt;&gt; Experiment type: ATAC-Seq</span></span>
<span><span class="co">#&gt; &gt;&gt; Matrix is filtered!</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; &gt;&gt; preparing features information...      2024-05-22 12:38:01 </span></span>
<span><span class="co">#&gt; &gt;&gt; identifying nearest features...        2024-05-22 12:38:02 </span></span>
<span><span class="co">#&gt; &gt;&gt; calculating distance from peak to TSS...   2024-05-22 12:38:02 </span></span>
<span><span class="co">#&gt; &gt;&gt; assigning genomic annotation...        2024-05-22 12:38:02 </span></span>
<span><span class="co">#&gt; &gt;&gt; assigning chromosome lengths           2024-05-22 12:38:11 </span></span>
<span><span class="co">#&gt; &gt;&gt; done...                    2024-05-22 12:38:11</span></span>
<span><span class="co">#&gt; &gt;&gt; Method: edgeR</span></span>
<span><span class="co">#&gt;  FDR:0.05&amp; abs(logFC)&lt;0</span></span>
<span><span class="co">#&gt; &gt;&gt; Estimating dispersion...</span></span>
<span><span class="co">#&gt; &gt;&gt; Fitting GLM...</span></span>
<span><span class="co">#&gt; &gt;&gt; DA peaks are found!</span></span>
<span><span class="co">#&gt; &gt;&gt; No `geneset` is specified so immune modules (Chaussabel, 2008) will be used!</span></span>
<span><span class="co">#&gt; &gt;&gt; enrichment.method` is not selected. Hyper-geometric p-value (HPEA) will be used!</span></span>
<span><span class="co">#&gt; &gt;&gt; Mice gene symbols are converted to human symbols!</span></span>
<span><span class="co">#&gt; &gt;&gt; Enrichment results are ready...</span></span>
<span><span class="co">#&gt; &gt;&gt; Done!</span></span>
<span></span>
<span><span class="fu"><a href="reference/pca_plot.html">pca_plot</a></span><span class="op">(</span><span class="va">results</span>, <span class="va">contrasts</span>, show.names <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/unnamed-chunk-5-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="single-cell-atac-seq-10x-scatac-preprocessing">Single-cell ATAC-seq (10x scATAC) preprocessing<a class="anchor" aria-label="anchor" href="#single-cell-atac-seq-10x-scatac-preprocessing"></a>
</h2>
<p>Use <code><a href="reference/prep_scATAC_cinaR.html">prep_scATAC_cinaR()</a></code> to pseudobulk 10x scATAC peak-by-cell matrices into a cinaR-ready consensus matrix.<br>
This preserves biological replicates (sample-level) and avoids inflated significance from per-cell testing.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># counts: peak-by-cell matrix (dense or dgCMatrix)</span></span>
<span><span class="co"># meta: data.frame with rownames = cell barcodes</span></span>
<span><span class="co"># meta must include biological replicate and condition columns</span></span>
<span></span>
<span><span class="va">prep</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/prep_scATAC_cinaR.html">prep_scATAC_cinaR</a></span><span class="op">(</span><span class="va">counts</span>, <span class="va">meta</span>,</span>
<span>                          sample.col <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>                          group.col <span class="op">=</span> <span class="st">"group"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">prep</span><span class="op">$</span><span class="va">bed</span>, <span class="va">prep</span><span class="op">$</span><span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"hg38"</span><span class="op">)</span></span></code></pre></div>
<p>Per-cell-type (sample × cluster) pseudobulk:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prep_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/prep_scATAC_cinaR.html">prep_scATAC_cinaR</a></span><span class="op">(</span><span class="va">counts</span>, <span class="va">meta</span>,</span>
<span>                               sample.col <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>                               group.col <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                               cluster.col <span class="op">=</span> <span class="st">"celltype"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">prep_list</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">bed</span>, <span class="va">x</span><span class="op">$</span><span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"hg38"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<p>Seurat/Signac object:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">prep</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/prep_scATAC_seurat.html">prep_scATAC_seurat</a></span><span class="op">(</span><span class="va">seurat_obj</span>,</span>
<span>                           sample.col <span class="op">=</span> <span class="st">"sample"</span>,</span>
<span>                           group.col <span class="op">=</span> <span class="st">"group"</span>,</span>
<span>                           assay <span class="op">=</span> <span class="st">"peaks"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">prep</span><span class="op">$</span><span class="va">bed</span>, <span class="va">prep</span><span class="op">$</span><span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"hg38"</span><span class="op">)</span></span></code></pre></div>
<p>If your peak IDs are not in <code>chr:start-end</code>, <code>chr_start_end</code>, or <code>chr-start-end</code> format, pass a <code>peak.bed</code> data.frame with <code>CHR</code>, <code>START</code>, and <code>STOP</code> columns via <code>peak.bed = ...</code>.</p>
<blockquote>
<p>For more details please go to our site from <a href="https://eonurk.github.io/cinaR/articles/cinaR.html">here!</a></p>
</blockquote>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="sc">@</span>article {Karakaslar2021.<span class="dv">03</span>.<span class="fl">05.434143</span>,</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>    author <span class="ot">=</span> {Karakaslar, E Onur and Ucar, Duygu},</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>    title <span class="ot">=</span> {cinaR<span class="sc">:</span> A comprehensive R package <span class="cf">for</span> the differential analyses and </span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>    functional interpretation of ATAC<span class="sc">-</span>seq data},</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>    year <span class="ot">=</span> {<span class="dv">2021</span>},</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>    doi <span class="ot">=</span> {<span class="fl">10.1101</span><span class="sc">/</span><span class="dv">2021</span>.<span class="dv">03</span>.<span class="fl">05.434143</span>},</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>    publisher <span class="ot">=</span> {Cold Spring Harbor Laboratory},</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>    URL <span class="ot">=</span> {https<span class="sc">:</span><span class="er">//</span>www.biorxiv.org<span class="sc">/</span>content<span class="sc">/</span><span class="fl">10.1101</span><span class="sc">/</span><span class="dv">2021</span>.<span class="dv">03</span>.<span class="fl">05.434143</span>v2},</span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>    journal <span class="ot">=</span> {bioRxiv}</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="contribution">Contribution<a class="anchor" aria-label="anchor" href="#contribution"></a>
</h2>
<p>You can send pull requests to make your contributions.</p>
</div>
<div class="section level2">
<h2 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h2>
<ul>
<li>GNU General Public License v3.0</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=cinaR" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/eonurk/cinaR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/eonurk/cinaR/issues/" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing cinaR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Onur Karakaslar <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-9182-910X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
<li>Duygu Ucar <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-9772-3066" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://cran.r-project.org/package=cinaR" class="external-link"><img src="https://www.r-pkg.org/badges/version/cinaR" alt="CRAN version"></a></li>
<li><a href="https://cran.r-project.org/package=cinaR?color=orange" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/cinaR?color=orange" alt="CRAN download"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Onur Karakaslar, Duygu Ucar.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
