<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>cinaR — cinaR • cinaR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="cinaR — cinaR"><meta name="description" content="Runs differential analyses and enrichment pipelines"><meta property="og:description" content="Runs differential analyses and enrichment pipelines"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">cinaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/cinaR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/eonurk/cinaR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>cinaR</h1>
      <small class="dont-index">Source: <a href="https://github.com/eonurk/cinaR/blob/master/R/cinaR.R" class="external-link"><code>R/cinaR.R</code></a></small>
      <div class="d-none name"><code>cinaR.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Runs differential analyses and enrichment pipelines</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">cinaR</span><span class="op">(</span></span>
<span>  <span class="va">matrix</span>,</span>
<span>  <span class="va">contrasts</span>,</span>
<span>  experiment.type <span class="op">=</span> <span class="st">"ATAC-Seq"</span>,</span>
<span>  DA.choice <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  DA.fdr.threshold <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  DA.lfc.threshold <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  comparison.scheme <span class="op">=</span> <span class="st">"OVO"</span>,</span>
<span>  save.DA.peaks <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  DA.peaks.path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  norm.method <span class="op">=</span> <span class="st">"cpm"</span>,</span>
<span>  filter.method <span class="op">=</span> <span class="st">"custom"</span>,</span>
<span>  library.threshold <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  cpm.threshold <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  TSS.threshold <span class="op">=</span> <span class="fl">50000</span>,</span>
<span>  show.annotation.pie <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  reference.genome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  batch.correction <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  batch.information <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  additional.covariates <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  sv.number <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  run.enrichment <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  enrichment.method <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  enrichment.FDR.cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  background.genes.size <span class="op">=</span> <span class="fl">20000</span>,</span>
<span>  geneset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-matrix">matrix<a class="anchor" aria-label="anchor" href="#arg-matrix"></a></dt>
<dd><p>either bed formatted consensus peak matrix (peaks by 3+samples)
CHR, START, STOP and raw peak counts OR count matrix (genes by 1+samples).</p></dd>


<dt id="arg-contrasts">contrasts<a class="anchor" aria-label="anchor" href="#arg-contrasts"></a></dt>
<dd><p>user-defined contrasts for comparing samples</p></dd>


<dt id="arg-experiment-type">experiment.type<a class="anchor" aria-label="anchor" href="#arg-experiment-type"></a></dt>
<dd><p>The type of experiment either set to "ATAC-Seq" or "RNA-Seq"</p></dd>


<dt id="arg-da-choice">DA.choice<a class="anchor" aria-label="anchor" href="#arg-da-choice"></a></dt>
<dd><p>determines which pipeline to run:
(1) edgeR, (2) limma-voom, (3) limma-trend, (4) DEseq2.
Note: Use limma-trend if consensus peaks are already normalized, otherwise use other methods.</p></dd>


<dt id="arg-da-fdr-threshold">DA.fdr.threshold<a class="anchor" aria-label="anchor" href="#arg-da-fdr-threshold"></a></dt>
<dd><p>fdr cut-off for differential analyses</p></dd>


<dt id="arg-da-lfc-threshold">DA.lfc.threshold<a class="anchor" aria-label="anchor" href="#arg-da-lfc-threshold"></a></dt>
<dd><p>log-fold change cutoff for differential analyses</p></dd>


<dt id="arg-comparison-scheme">comparison.scheme<a class="anchor" aria-label="anchor" href="#arg-comparison-scheme"></a></dt>
<dd><p>either one-vs-one (OVO) or one-vs-all (OVA) comparisons.</p></dd>


<dt id="arg-save-da-peaks">save.DA.peaks<a class="anchor" aria-label="anchor" href="#arg-save-da-peaks"></a></dt>
<dd><p>saves differentially accessible peaks to an excel file</p></dd>


<dt id="arg-da-peaks-path">DA.peaks.path<a class="anchor" aria-label="anchor" href="#arg-da-peaks-path"></a></dt>
<dd><p>the path which the excel file of the DA peaks will be saved,
if not set it will be saved to current directory.</p></dd>


<dt id="arg-norm-method">norm.method<a class="anchor" aria-label="anchor" href="#arg-norm-method"></a></dt>
<dd><p>normalization method for consensus peaks</p></dd>


<dt id="arg-filter-method">filter.method<a class="anchor" aria-label="anchor" href="#arg-filter-method"></a></dt>
<dd><p>filtering method for low expressed peaks</p></dd>


<dt id="arg-library-threshold">library.threshold<a class="anchor" aria-label="anchor" href="#arg-library-threshold"></a></dt>
<dd><p>number of libraries a peak occurs so that it is not filtered default set to 2</p></dd>


<dt id="arg-cpm-threshold">cpm.threshold<a class="anchor" aria-label="anchor" href="#arg-cpm-threshold"></a></dt>
<dd><p>count per million threshold for not to filter a peak</p></dd>


<dt id="arg-tss-threshold">TSS.threshold<a class="anchor" aria-label="anchor" href="#arg-tss-threshold"></a></dt>
<dd><p>Distance to transcription start site in base-pairs. Default set to 50,000.</p></dd>


<dt id="arg-show-annotation-pie">show.annotation.pie<a class="anchor" aria-label="anchor" href="#arg-show-annotation-pie"></a></dt>
<dd><p>shows the annotation pie chart produced with ChipSeeker</p></dd>


<dt id="arg-reference-genome">reference.genome<a class="anchor" aria-label="anchor" href="#arg-reference-genome"></a></dt>
<dd><p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p></dd>


<dt id="arg-batch-correction">batch.correction<a class="anchor" aria-label="anchor" href="#arg-batch-correction"></a></dt>
<dd><p>logical, if set will run unsupervised batch correction
via sva (default) or if the batch information is known `batch.information`
argument should be provided by user.</p></dd>


<dt id="arg-batch-information">batch.information<a class="anchor" aria-label="anchor" href="#arg-batch-information"></a></dt>
<dd><p>character vector, given by user.</p></dd>


<dt id="arg-additional-covariates">additional.covariates<a class="anchor" aria-label="anchor" href="#arg-additional-covariates"></a></dt>
<dd><p>vector or data.frame, this parameter will be directly added to design
matrix before running the differential analyses, therefore won't affect the batch corrections but
adjust the results in down-stream analyses.</p></dd>


<dt id="arg-sv-number">sv.number<a class="anchor" aria-label="anchor" href="#arg-sv-number"></a></dt>
<dd><p>number of surrogate variables to be calculated using SVA, best left untouched.</p></dd>


<dt id="arg-run-enrichment">run.enrichment<a class="anchor" aria-label="anchor" href="#arg-run-enrichment"></a></dt>
<dd><p>logical, turns off enrichment pipeline</p></dd>


<dt id="arg-enrichment-method">enrichment.method<a class="anchor" aria-label="anchor" href="#arg-enrichment-method"></a></dt>
<dd><p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p></dd>


<dt id="arg-enrichment-fdr-cutoff">enrichment.FDR.cutoff<a class="anchor" aria-label="anchor" href="#arg-enrichment-fdr-cutoff"></a></dt>
<dd><p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p></dd>


<dt id="arg-background-genes-size">background.genes.size<a class="anchor" aria-label="anchor" href="#arg-background-genes-size"></a></dt>
<dd><p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p></dd>


<dt id="arg-geneset">geneset<a class="anchor" aria-label="anchor" href="#arg-geneset"></a></dt>
<dd><p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using `read.gmt` function from `qusage`
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>prints messages through running the pipeline</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>returns differentially accessible peaks</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">atac_seq_consensus_bm</span><span class="op">)</span> <span class="co"># calls 'bed'</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># a vector for comparing the examples</span></span></span>
<span class="r-in"><span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span>, split <span class="op">=</span> <span class="st">"-"</span>, fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">}</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">25</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">cinaR</span><span class="op">(</span><span class="va">bed</span>, <span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"mm10"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Experiment type: ATAC-Seq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Matrix is filtered!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; preparing features information...		 2026-02-02 12:19:22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; Using Genome: mm10 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; identifying nearest features...		 2026-02-02 12:19:22 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; calculating distance from peak to TSS...	 2026-02-02 12:19:23 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; assigning genomic annotation...		 2026-02-02 12:19:23 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; Using Genome: mm10 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; Using Genome: mm10 ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; assigning chromosome lengths			 2026-02-02 12:19:38 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; done...					 2026-02-02 12:19:38 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Method: edgeR</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	FDR:0.05&amp; abs(logFC)&lt;0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Estimating dispersion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Fitting GLM...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; DA peaks are found!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; No `geneset` is specified so immune modules (Chaussabel, 2008) will be used!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; enrichment.method` is not selected. Hyper-geometric p-value (HPEA) will be used!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Mice gene symbols are converted to human symbols!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Enrichment results are ready...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Done!</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Onur Karakaslar, Duygu Ucar.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

