<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>run_enrichment — run_enrichment • cinaR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="run_enrichment — run_enrichment"><meta name="description" content="This function is run, if the enrichment pipeline wants to be called afterwards.
Setting reference genome to the same genome which cinaR was run should be given to this function!"><meta property="og:description" content="This function is run, if the enrichment pipeline wants to be called afterwards.
Setting reference genome to the same genome which cinaR was run should be given to this function!"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">cinaR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/cinaR.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/eonurk/cinaR/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>run_enrichment</h1>
      <small class="dont-index">Source: <a href="https://github.com/eonurk/cinaR/blob/master/R/enrichment_functions.R" class="external-link"><code>R/enrichment_functions.R</code></a></small>
      <div class="d-none name"><code>run_enrichment.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function is run, if the enrichment pipeline wants to be called afterwards.
Setting reference genome to the same genome which cinaR was run should be given to this function!</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_enrichment</span><span class="op">(</span></span>
<span>  <span class="va">results</span>,</span>
<span>  geneset <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  experiment.type <span class="op">=</span> <span class="st">"ATAC-Seq"</span>,</span>
<span>  reference.genome <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  enrichment.method <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  enrichment.FDR.cutoff <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  background.genes.size <span class="op">=</span> <span class="fl">20000</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-results">results<a class="anchor" aria-label="anchor" href="#arg-results"></a></dt>
<dd><p>list, DA peaks list for different contrasts</p></dd>


<dt id="arg-geneset">geneset<a class="anchor" aria-label="anchor" href="#arg-geneset"></a></dt>
<dd><p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using `read.gmt` function from `qusage`
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p></dd>


<dt id="arg-experiment-type">experiment.type<a class="anchor" aria-label="anchor" href="#arg-experiment-type"></a></dt>
<dd><p>The type of experiment either set to "ATAC-Seq" or "RNA-Seq"</p></dd>


<dt id="arg-reference-genome">reference.genome<a class="anchor" aria-label="anchor" href="#arg-reference-genome"></a></dt>
<dd><p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p></dd>


<dt id="arg-enrichment-method">enrichment.method<a class="anchor" aria-label="anchor" href="#arg-enrichment-method"></a></dt>
<dd><p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p></dd>


<dt id="arg-enrichment-fdr-cutoff">enrichment.FDR.cutoff<a class="anchor" aria-label="anchor" href="#arg-enrichment-fdr-cutoff"></a></dt>
<dd><p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p></dd>


<dt id="arg-background-genes-size">background.genes.size<a class="anchor" aria-label="anchor" href="#arg-background-genes-size"></a></dt>
<dd><p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>prints messages through running the pipeline</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>list, enrichment analyses results along with corresponding differential analyses outcomes</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/eonurk/cinaR/" class="external-link">cinaR</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">atac_seq_consensus_bm</span><span class="op">)</span> <span class="co"># calls 'bed'</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># a vector for comparing the examples</span></span></span>
<span class="r-in"><span><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span>, split <span class="op">=</span> <span class="st">"-"</span>, fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">}</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">25</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="cinaR.html">cinaR</a></span><span class="op">(</span><span class="va">bed</span>, <span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"mm10"</span>, run.enrichment <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Experiment type: ATAC-Seq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Matrix is filtered!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Package "ChIPseeker" needed for this function to work. Please install it.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in abs(cp.filtered.annotated$distanceToTSS):</span> non-numeric argument to mathematical function</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">results_with_enrichment</span> <span class="op">&lt;-</span> <span class="fu">run_enrichment</span><span class="op">(</span><span class="va">results</span>, reference.genome <span class="op">=</span> <span class="st">"mm10"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error:</span> object 'results' not found</span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Onur Karakaslar, Duygu Ucar.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer></div>





  </body></html>

