<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>cinaR — cinaR • cinaR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/sandstone/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="cinaR — cinaR"><meta property="og:description" content="Runs differential analyses and enrichment pipelines"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-116646865-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-116646865-1');
</script></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cinaR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.2.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/cinaR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/eonurk/cinaR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>cinaR</h1>
    <small class="dont-index">Source: <a href="https://github.com/eonurk/cinaR/blob/HEAD/R/cinaR.R" class="external-link"><code>R/cinaR.R</code></a></small>
    <div class="hidden name"><code>cinaR.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Runs differential analyses and enrichment pipelines</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">cinaR</span><span class="op">(</span>
  <span class="va">matrix</span>,
  <span class="va">contrasts</span>,
  experiment.type <span class="op">=</span> <span class="st">"ATAC-Seq"</span>,
  DA.choice <span class="op">=</span> <span class="fl">1</span>,
  DA.fdr.threshold <span class="op">=</span> <span class="fl">0.05</span>,
  DA.lfc.threshold <span class="op">=</span> <span class="fl">0</span>,
  comparison.scheme <span class="op">=</span> <span class="st">"OVO"</span>,
  save.DA.peaks <span class="op">=</span> <span class="cn">FALSE</span>,
  DA.peaks.path <span class="op">=</span> <span class="cn">NULL</span>,
  norm.method <span class="op">=</span> <span class="st">"cpm"</span>,
  filter.method <span class="op">=</span> <span class="st">"custom"</span>,
  library.threshold <span class="op">=</span> <span class="fl">2</span>,
  cpm.threshold <span class="op">=</span> <span class="fl">1</span>,
  TSS.threshold <span class="op">=</span> <span class="fl">50000</span>,
  show.annotation.pie <span class="op">=</span> <span class="cn">FALSE</span>,
  reference.genome <span class="op">=</span> <span class="cn">NULL</span>,
  batch.correction <span class="op">=</span> <span class="cn">FALSE</span>,
  batch.information <span class="op">=</span> <span class="cn">NULL</span>,
  additional.covariates <span class="op">=</span> <span class="cn">NULL</span>,
  sv.number <span class="op">=</span> <span class="cn">NULL</span>,
  run.enrichment <span class="op">=</span> <span class="cn">TRUE</span>,
  enrichment.method <span class="op">=</span> <span class="cn">NULL</span>,
  enrichment.FDR.cutoff <span class="op">=</span> <span class="fl">1</span>,
  background.genes.size <span class="op">=</span> <span class="fl">20000</span>,
  geneset <span class="op">=</span> <span class="cn">NULL</span>,
  verbose <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>matrix</dt>
<dd><p>either bed formatted consensus peak matrix (peaks by 3+samples)
CHR, START, STOP and raw peak counts OR count matrix (genes by 1+samples).</p></dd>
<dt>contrasts</dt>
<dd><p>user-defined contrasts for comparing samples</p></dd>
<dt>experiment.type</dt>
<dd><p>The type of experiment either set to "ATAC-Seq" or "RNA-Seq"</p></dd>
<dt>DA.choice</dt>
<dd><p>determines which pipeline to run:
(1) edgeR, (2) limma-voom, (3) limma-trend, (4) DEseq2.
Note: Use limma-trend if consensus peaks are already normalized, otherwise use other methods.</p></dd>
<dt>DA.fdr.threshold</dt>
<dd><p>fdr cut-off for differential analyses</p></dd>
<dt>DA.lfc.threshold</dt>
<dd><p>log-fold change cutoff for differential analyses</p></dd>
<dt>comparison.scheme</dt>
<dd><p>either one-vs-one (OVO) or one-vs-all (OVA) comparisons.</p></dd>
<dt>save.DA.peaks</dt>
<dd><p>saves differentially accessible peaks to an excel file</p></dd>
<dt>DA.peaks.path</dt>
<dd><p>the path which the excel file of the DA peaks will be saved,
if not set it will be saved to current directory.</p></dd>
<dt>norm.method</dt>
<dd><p>normalization method for consensus peaks</p></dd>
<dt>filter.method</dt>
<dd><p>filtering method for low expressed peaks</p></dd>
<dt>library.threshold</dt>
<dd><p>number of libraries a peak occurs so that it is not filtered default set to 2</p></dd>
<dt>cpm.threshold</dt>
<dd><p>count per million threshold for not to filter a peak</p></dd>
<dt>TSS.threshold</dt>
<dd><p>Distance to transcription start site in base-pairs. Default set to 50,000.</p></dd>
<dt>show.annotation.pie</dt>
<dd><p>shows the annotation pie chart produced with ChipSeeker</p></dd>
<dt>reference.genome</dt>
<dd><p>genome of interested species. It should be 'hg38', 'hg19' or 'mm10'.</p></dd>
<dt>batch.correction</dt>
<dd><p>logical, if set will run unsupervised batch correction
via sva (default) or if the batch information is known `batch.information`
argument should be provided by user.</p></dd>
<dt>batch.information</dt>
<dd><p>character vector, given by user.</p></dd>
<dt>additional.covariates</dt>
<dd><p>vector or data.frame, this parameter will be directly added to design
matrix before running the differential analyses, therefore won't affect the batch corrections but
adjust the results in down-stream analyses.</p></dd>
<dt>sv.number</dt>
<dd><p>number of surrogate variables to be calculated using SVA, best left untouched.</p></dd>
<dt>run.enrichment</dt>
<dd><p>logical, turns off enrichment pipeline</p></dd>
<dt>enrichment.method</dt>
<dd><p>There are two methodologies for enrichment analyses,
Hyper-geometric p-value (HPEA) or Geneset Enrichment Analyses (GSEA).</p></dd>
<dt>enrichment.FDR.cutoff</dt>
<dd><p>FDR cut-off for enriched terms, p-values
are corrected by Benjamini-Hochberg procedure</p></dd>
<dt>background.genes.size</dt>
<dd><p>number of background genes for hyper-geometric p-value
calculations. Default is 20,000.</p></dd>
<dt>geneset</dt>
<dd><p>Pathways to be used in enrichment analyses. If not set vp2008 (Chaussabel, 2008)
immune modules will be used. This can be set to any geneset using `read.gmt` function from `qusage`
package. Different modules are available: https://www.gsea-msigdb.org/gsea/downloads.jsp.</p></dd>
<dt>verbose</dt>
<dd><p>prints messages through running the pipeline</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>returns differentially accessible peaks</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co"># \donttest{</span></span>
<span class="r-in"><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">atac_seq_consensus_bm</span><span class="op">)</span> <span class="co"># calls 'bed'</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># a vector for comparing the examples</span></span>
<span class="r-in"><span class="va">contrasts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/strsplit.html" class="external-link">strsplit</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bed</span><span class="op">)</span>, split <span class="op">=</span> <span class="st">"-"</span>, fixed <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span class="r-in">                    <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span><span class="va">x</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">}</span><span class="op">)</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">25</span><span class="op">]</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">cinaR</span><span class="op">(</span><span class="va">bed</span>, <span class="va">contrasts</span>, reference.genome <span class="op">=</span> <span class="st">"mm10"</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Experiment type: ATAC-Seq</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Matrix is filtered!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; preparing features information...		 2022-05-06 16:30:12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; identifying nearest features...		 2022-05-06 16:30:12 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; calculating distance from peak to TSS...	 2022-05-06 16:30:13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; assigning genomic annotation...		 2022-05-06 16:30:13 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; assigning chromosome lengths			 2022-05-06 16:30:27 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> &gt;&gt; done...					 2022-05-06 16:30:27 </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Method: edgeR</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	FDR:0.05&amp; abs(logFC)&lt;0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Estimating dispersion...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Fitting GLM...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; DA peaks are found!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; No `geneset` is specified so immune modules (Chaussabel, 2008) will be used!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; enrichment.method` is not selected. Hyper-geometric p-value (HPEA) will be used!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Mice gene symbols are converted to human symbols!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Enrichment results are ready...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> &gt;&gt; Done!</span>
<span class="r-in"><span class="co"># }</span></span>
<span class="r-in"></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by <a href="http://eonurk.github.io" class="external-link">Onur Karakaslar</a>, Duygu Ucar.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

      </footer></div>

  


  

  </body></html>

